enforcer-parent/enforcer/src/main/java/org/wso2/choreo/connect/enforcer/api/RestAPI.java

```diff
@@ -165,9 +166,13 @@ public ResponseObject process(RequestContext requestContext) {
             String authHeaderName = FilterUtils.getAuthHeaderName(requestContext);
             requestContext.getRemoveHeaders().add(authHeaderName);
         }
+        // Authorization Header should not be included in the throttle publishing event.
+        requestContext.getProtectedHeaders().add(FilterUtils.getAuthHeaderName(requestContext));
```

Comment:
why can't we just use removeHeaders for this purpose?

Reply:
Because if the user enables forwarding authorization header to the Backend, it would be published to the throttle event